modules/tide_oauth/drush/tide_oauth.drush.inc

```diff
@@ -33,4 +37,15 @@ function drush_tide_oauth_env_keygen() {
       drush_set_error('Could not generate OAuth keys from TIDE_OAUTH_*_KEY environment variables.');
     }
   }
+  else {
+    /** @var \Drupal\simple_oauth\Service\KeyGeneratorService $key_generator */
+    $key_generator = \Drupal::service('simple_oauth.key.generator');
```

Comment:
@MdNadimHossain @anthony-malkoun I think those lines should come as a public method of `tide_oauth.env_key_generator` service to reduce the duplication in both Drush 8 and Drush 9 commands and also to keep the Drush command minimal.

Probably just `$env_key_generator->generateKeys()` for the logic of the whole command:
* check for `$this->hasEnvKeys()` and call `$this->generateEnvKeys()`
* otherwise call another method to generate keys via `$key_generator->generateKeys('private://');`.

Reply:
Thanks, @sonnykt I have cleaned up the code 👍